ludwig/features/feature_utils.py

```diff
@@ -100,23 +100,19 @@ def sanitize(name):
 
 
 def compute_feature_hash(feature: dict) -> str:
-    """
-    This function computes a hash for each feature based on the preprocessing dictionary associated with each feature.
-    The input is always the feature dict, however sometimes, this is called from BaseFeature which dumps the feature
-    dict from a ConfigObject and the preprocessing key corresponds to a nested Preprocessing config. This is why it
-    includes the if/else statement.
+    """This function computes a hash for each feature based on the preprocessing dictionary associated with each
```

Comment:
This function is only called once after the `ModelConfig` is constructed. As such, instead of calling `feature.to_dict()` you can pass the feature object directly in, which makes it more type safe and avoids the unnecessary safety check when getting the PREPROCESSING section (with the ModelConfig, the preprocessing section is always populated).

Reply:
I started making this change, but realized that our tests have a deep relationship with `compute_feature_hash` and are super sensitive to this. Too many test failures downstream locally, so I think we need to refactor our tests to not use dicts.

For e.g., even something as simple as `text_feature()` and `number_feature()` initialization in our tests compute the proc hashes in their implementations, and getting rid of them causes other errors because many of our tests directly use/reference the proc hash columns as dicts rather than attributes.

I'll skip this for now, but maybe this is something we should also follow up with.

Reply:
Ah, I see! Yes, this is a bigger refactor we'll need to make.